src/libraries/Locker.sol

```diff
@@ -14,31 +14,25 @@ library Locker {
     /// we use locker==address(0) to signal that the pool is not locked
     error InvalidLocker();
 
-    function setLocker(address locker) internal {
-        if (locker == address(0)) revert InvalidLocker();
+    function lock() internal {
         uint256 slot = LOCKER_SLOT;
```

Comment:
nit: we should rename `LOCKER_SLOT`... maybe `LOCKED_SLOT` or ... `LOCK_SLOT` ? Open to other ideas

Reply:
I like LOCK_SLOT